source/application/AMPIC887DriverServer/AMPIC887ConsoleApplication.cpp

```diff
@@ -70,23 +66,507 @@ void AMPIC887ConsoleApplication::onActiveControllerChangeIssued(const QString &c
 	}
 }
 
-void AMPIC887ConsoleApplication::onGCS2CommandIssued(AMGCS2Command* command)
+void AMPIC887ConsoleApplication::onControllerOutputReceived(const QString &outputString)
 {
-	if(!controllerCollection_.activeController()) {
+	consoleInputHandler_->writeLineToStandardOutput(outputString);
+}
+
+void AMPIC887ConsoleApplication::onControllerErrorEncountered(const QString &errorMessage)
+{
+	consoleInputHandler_->writeLineToStandardError(errorMessage);
+}
+
+void AMPIC887ConsoleApplication::onSetCommandLevelCommandIssued(AMGCS2::ControllerCommandLevel commandLevel,
+																const QString &password)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
```

Comment:
I believe I mentioned somewhere else recently.  I prefer positive conditional formatting that doesn't use returns to stop bad input/state access.  Therefore this method would be rewritten as:

```c++
AMPIC887Controller *activeController = controllerCollection_.activeController();

if (activeController && !activeController->setCommandLevel(commandLevel, password))
     consoleInputHandler_->writeLineToStandardError(activeController->lastError());
```